### PR TITLE
Ensure no pre-allocation occurs when --reuse-alloc not specified

### PR DESCRIPTION
**Pull request by pentschev**
Fixes the behavior of benchmarks when `--reuse-alloc` is not specified, before this change objects would be pre-allocated, even though that didn't guarantee the underlying memory would be allocated before running the transfer. Only `--reuse-alloc` should pre-allocate objects or buffers.

### CODE REVIEW

File: ucp/benchmarks/backends/ucp_async.py [hunk 1]
```diff
@@ -131,6 +127,10 @@ async def run(self):
                 await ep.am_send(msg)
                 await ep.am_recv()
             else:
+                if not self.args.reuse_alloc:
+                    msg_send_list.append(self.xp.arange(self.args.n_bytes, dtype="u1"))
+                    msg_recv_list.append(self.xp.zeros(self.args.n_bytes, dtype="u1"))
+
```
[the remark this answers is not in the document]
Comment (wence-) [reply ucp/benchmarks/backends/ucp_async.py:133]: Rationale for this is also that we don't really look at the results of all the `msg_send_list` etc...

**Review by wence- — COMMENTED** [review]
Possible simplification?

File: ucp/benchmarks/backends/ucp_async.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pentschev) [reply ucp/benchmarks/backends/ucp_async.py]: I tried to combine the changes you suggested here in https://github.com/rapidsai/ucx-py/pull/880/commits/3e748b8b403cb1cdfc59461694156e1d4aa9e2ef, could you take another look?

**Review by wence- — CHANGES_REQUESTED** [review]
Sorry, I realise I produced a load of bad advice in the last round of comments (which most of my suggestions are trying to undo now). That aside, this change looks overall good, thanks!

**Comment by pentschev** [comment]
> Sorry, I realise I produced a load of bad advice in the last round of comments (which most of my suggestions are trying to undo now). That aside, this change looks overall good, thanks!

Your new comments are correct I believe. No worries, I should also have been more attentive while I was doing those changes, I've been distracted and overlooked the `gather` issues, even though we've had those before. These are the times I would love a big red failure, unfortunate data races are much sneakier. 🙂

**Review by wence- — APPROVED** [review]
Thanks, looks great!

**Pull request merged**